### PR TITLE
Tools ml details

### PR DESCRIPTION
**Pull request by jordan2lee**
1. Add tool for users to explore containerized models (algorithm, model parameters, reduced feature list)
2. Update wording of docs
3. Remove synapse id’s that are no longer needed (can download at publication page)
4. Updated algorithm data
5. Handle for quantile negatives with flooring to 0
6. Updated details on each algorithm method
7. Updated docs and invocation of CloudForest for latest version of CloudForest
8. Renamed tooling

Includes several PRs
PR #51 handles when 10Q TMP lower than user input
PR #52 Cf.v2
PR #53 automated consolidation of all ML method results

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,13 +110,14 @@ AKLIMATE and subSCOPE do not need manual model data download.
...
+The `ft_name_convert.tar.gz` file can be downloaded from the Publication Page and then placed in `tools/`
 ```
 cd tools
 tar -xzf ft_name_convert.tar.gz
 cd ..
 ```
 
+
 # Data Requirements
 User input data must be in tab separated format.
```
Comment (briankarlberg) [inline README.md:122]: Samples as rows, features as columns; first column is Labels.

File: README.md
Comment (briankarlberg) [inline README.md:1]: well-performing (add dash)

File: tools/cancer2subtype.json
Comment (briankarlberg) [inline tools/cancer2subtype.json:1]: I don't recall any cancer types types having 8 subtypes as with GEA and PRAD. I think KICHKIRC is KIRCKICH. Is there a way to automatically regenerate this?

File: tools/tools_ml.py
Comment (briankarlberg) [inline tools/tools_ml.py:1]: ACC is out of order (?)

File: tools/format_results.py
Comment (briankarlberg) [inline tools/format_results.py:1]: voted-for

File: tools/format_results.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+first = 1
+for method,df in results_d.items():
+    if first == 1:
+        base = df
+        first = 0
+    else:
+        base = pd.merge(base, df, on=['sampleID', 'TCGA_cohort', 'platform'], how='outer')
+
+# all absolute sample subtype call column
+# this shows the most voted for subtype
```
Comment (briankarlberg) [inline tools/format_results.py:29]: voted-for

File: tools/tools_ml.py [hunk 3]
```diff
@@ -27,3 +27,19 @@ def gexp_converter(data_entrez_list, cancer):
...
+
+def get_model_info(method, platform, cancer):
+    '''retrieves TMP model information
+    includes: model name, model parameters, and feature list (reported in TMP nomenclature)
+
+    specify which model you would like to look at. example: top performing jadbio gene expression only model
+
+    options method = ['skgrid', 'aklimate', 'cloudforest', 'subscope', 'jadbio']
+    options platform = ['GEXP', 'CNVR', 'MIR', 'MUTA', 'METH']
+    options cancer = ['BRCA', 'LGGGBM', 'COADREAD', 'SKCM', 'ACC', 'BLCA', 'CESC', 'ESCC', 'GEA', 'HNSC', 'KIRCKICH', 'KIRP', 'LIHCCHOL', 'LUAD', 'LUSC', 'MESO', 'OV', 'PAAD', 'PCPG', 'PRAD', 'SARC', 'TGCT', 'THCA', 'THYM', 'UCEC', 'UVM']
```
Comment (briankarlberg) [inline tools/tools_ml.py:39]: ACC out of order

File: tutorial/Explore_models.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+### Allowed Options
+`get_model_info(method, platform, cancer)` allows for:
+
+Options `method` are ['skgrid', 'aklimate', 'cloudforest', 'subscope', 'jadbio']
+
+Options `platform` are ['GEXP', 'CNVR', 'MIR', 'MUTA', 'METH', 'OVERALL', 'All']. Note 'OVERALL' and 'All' are for CloudForest only
+
+Options `cancer` are ['BRCA', 'LGGGBM', 'COADREAD', 'SKCM', 'ACC', 'BLCA', 'CESC', 'ESCC', 'GEA', 'HNSC', 'KIRCKICH', 'KIRP', 'LIHCCHOL', 'LUAD', 'LUSC', 'MESO', 'OV', 'PAAD', 'PCPG', 'PRAD', 'SARC', 'TGCT', 'THCA', 'THYM', 'UCEC', 'UVM']
```
Comment (briankarlberg) [inline tutorial/Explore_models.md:40]: ACC, SKCM out of order

File: README.md (hunk 1 above)
Comment (jordan2lee) [reply README.md]: good point, updated in commit [d24da3e](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/d24da3ef436a4ee4040cdaf53c2c2bdd3532c46c)

File: README.md
Comment (jordan2lee) [reply README.md:1]: good point, updated in commit [d24da3e](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/d24da3ef436a4ee4040cdaf53c2c2bdd3532c46c)

File: tools/tools_ml.py (hunk 3 above)
Comment (jordan2lee) [reply tools/tools_ml.py]: updated the order of cancer cohorts in commit [9b09147](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/9b09147d35ceab6a336723b6c74ef06ac86c0ce2)

File: tools/format_results.py
Comment (jordan2lee) [reply tools/format_results.py:1]: updated wording in commit [98aea06](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/98aea06b0e0548ac6035507af4ca47a35004e2f6)

File: tools/format_results.py (hunk 2 above)
Comment (jordan2lee) [reply tools/format_results.py]: updated in commit [9b09147](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/9b09147d35ceab6a336723b6c74ef06ac86c0ce2)

File: tutorial/Explore_models.md (hunk 4 above)
Comment (jordan2lee) [reply tutorial/Explore_models.md]: good eye, updated in commit [f2ecb03](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/f2ecb033d1c5059f8ea7b23ae02fdd32be24a80b)

File: tools/tools_ml.py
Comment (jordan2lee) [reply tools/tools_ml.py:1]: updated in commit [9b09147](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/9b09147d35ceab6a336723b6c74ef06ac86c0ce2)

File: tools/cancer2subtype.json
Comment (jordan2lee) [reply tools/cancer2subtype.json:1]: good eye, I'll dive into this and update. Thanks BK!

Comment (jordan2lee) [reply tools/cancer2subtype.json:1]: updated in commits 

[9eaa6e3](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/9eaa6e327d9bfb58e881ae65ae915f3d6f43c1e9)

[6aeb9c7](https://github.com/NCICCGPO/gdan-tmp-models/pull/54/commits/6aeb9c786dbead78c4ffa8ee460dca6f7bd59c5e)

**Pull request merged**